gematria/datasets/process_and_filter_bbs.cc

```diff
@@ -79,6 +82,7 @@ Expected<std::string> ProcessBasicBlock(
        *DisassembledInstructionsOrErr) {
     MCInstrDesc InstDesc =
         LLVMSupport.mc_instr_info().get(Instruction.mc_inst.getOpcode());
+    if (Instruction.mc_inst.getOpcode() == X86::SYSCALL) continue;
```

Comment:
There is a whole category of instructions that may have side-effects that are not modelled by the other fields of the descriptor. We probably want to skip all of those.

See https://github.com/llvm-mirror/llvm/blob/2c4ca6832fa6b306ee6a7010bfb80a3f2596f824/include/llvm/MC/MCInstrDesc.h#L443 for more details.

Reply:
Good point. I didn't realize that was a field of `MCInstDesc`. I've added a todo for now as I know filtering syscalls works to reasonable effectiveness and am not sure what the consequences of filtering everything with non-modelled side effects is at scale.